src/probestack.rs

```diff
@@ -43,18 +43,32 @@
 
 #![cfg(not(windows))] // Windows already has builtins to do this
 
+extern "C" {
+    pub fn __rust_probestack();
+}
+
 #[naked]
 #[no_mangle]
 #[cfg(all(target_arch = "x86_64", not(feature = "mangled-names")))]
-pub unsafe extern "C" fn __rust_probestack() {
+pub unsafe extern "C" fn __rust_probestack_wrapper() {
     // Our goal here is to touch each page between %rsp+8 and %rsp+8-%rax,
     // ensuring that if any pages are unmapped we'll make a page fault.
     //
     // The ABI here is that the stack frame size is located in `%eax`. Upon
     // return we're not supposed to modify `%esp` or `%eax`.
     asm!("
+        .cfi_endproc
+        .pushsection .text.__rust_probestack
+        .p2align 4,,15
+        .globl __rust_probestack
+        .type  __rust_probestack, @function
+    __rust_probestack:
+        .cfi_startproc
         pushq  %rbp
+        .cfi_def_cfa_offset 16
```

Comment:
I'd use `.cfi_adjust_cfa_offset 8` here--that exactly matches the change made by the push insn, rather than freshly encoding the absolute state.